### PR TITLE
Transfer proof

### PR DESCRIPTION
**Pull request by mrsmkl**
## Description

Implement the condition and agreement for access with transfer proof.

Currently the condition is pretty generic, so just access to some data perhaps should be connected to DID hash or something?

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

**Comment by mrsmkl** [comment]
Here's the POC for handling challenges https://github.com/nevermined-io/data-transfer-poc

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -16,16 +16,19 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v1
         with:
-          node-version: '10.x'
```
Comment (aaitor) [inline .github/workflows/build.yml:19]: I'm not sure about changing the node version here. Previous experiences bumping to node >=12.x created some issues.
Is node >=14.x a requirement?

File: circuits/build.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Comment (aaitor) [inline circuits/build.sh:1]: Maybe move this one to the `scripts` folder

File: contracts/conditions/AccessProofCondition.sol [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+import '../interfaces/ISecretStore.sol';
+import '../interfaces/ISecretStorePermission.sol';
+import '../agreements/AgreementStoreManager.sol';
+
+interface IDisputeManager {
+    function verifyProof(bytes memory proof, uint[] memory pubSignals) external view returns (bool);
+}
+
+/**
+ * @title Access Condition with transfer proof
```
Comment (aaitor) [inline contracts/conditions/AccessProofCondition.sol:18]: Update doc

File: contracts/conditions/AccessProofCondition.sol [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    )
+        public
+        pure
+        returns (bytes32)
+    {
+        return keccak256(abi.encode(_origHash, _buyer[0], _buyer[1], _provider[0], _provider[1]));
+    }
+
+   /**
+    * @notice fulfill access secret store condition
```
Comment (aaitor) [inline contracts/conditions/AccessProofCondition.sol:97]: update doc

**Review by aaitor — APPROVED** [review]
lgtm

File: .github/workflows/build.yml (hunk 1 above)
Comment (mrsmkl) [reply .github/workflows/build.yml:19]: yeah, looks like snarkjs and circom depend on it...

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@nevermined-io/contracts",
-  "version": "1.1.2",
+  "version": "1.1.3",
```
Comment (aaitor) [inline package.json:3]: last tag already released is 1.1.4 so it should be 1.1.5

Comment (aaitor) [reply package.json:3]: version needs to change in the `pom.xml` and `setup.py` files too in order to generate the maven and pypi artifacts

**Comment by aaitor** [comment]
:shipit:

**Pull request merged**